### DISCUSSION TITLE
🙏 Help: List of CustomElements

### DISCUSSION THREAD

**Discussion by deleonio** [public-ui/kolibri#6544] [Q&A]
### Summary

Do you already have a function that can return all the names of the available CustomElements in a string list? Or would it be better to write them ourselves from the `custom-elements.json` file? If the latter, how stable is this file?

**Comment by deleonio** [comment]
Exactly. The `custom-elements.json` is generated when building KoliBri and is located in the NPM package `@public-ui/components` ([NPM-Content](https://www.npmjs.com/package/@public-ui/components?activeTab=code)). There it can always be read and processed in versioned form.

FYI: @Chrisdo82

**Reply by Chrisdo82** [reply]
Thank you

**Reply by Chrisdo82** [reply]
@deleonio 

This is not exactly what I need. To clarify my question, what I want to have is a TS Union Type consisting of all the custom element names:

`'kol-abbr' | 'kol-accordion' | 'kol-alert' | ...`

It's not possible with Typescript to extract the type of a json file. 

```ts
import koliBriCustomElementsDefinition from '@public-ui/components/custom-elements.json' assert { type: 'json' }

const names = koliBriCustomElementsDefinition.tags.map((tag) => tag.name)
type CustomElementNames = typeof names[number]
```

this results in `type CustomElementNames = string`

**Comment by deleonio** [comment]
Hier geht es darum, dass man die Component-Types wiederverwenden kann: https://github.com/public-ui/kolibri/issues/8613
